lib/index.js

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+var _ = require('lodash');
+var async = require('async');
+var bedrock = require('bedrock');
+var crypto = require('crypto');
+var jwt = require('jsonwebtoken');
+var database = require('bedrock-mongodb');
+var BedrockError = bedrock.util.BedrockError;
+
+var _state = null;
+
+// load config defaults
+require('./config');
+
+var api = {};
+module.exports = api;
+
+var logger = bedrock.loggers.get('app');
+
+// TODO: make responding to this event optional
+bedrock.events.on(
+  'bedrock-authn-did-jwt.config.secretStore', (strategy, callback) => {
+    strategy.setStore(api, callback);
+  });
+
+bedrock.events.on('bedrock-mongodb.ready', callback => {
+  async.waterfall([
+    callback => {
+      database.openCollections(['jwtSecrets'], callback);
```

Comment:
I see you used `key store` in some of the other comments... `jwtKeyStore` would also be an ok name, your call.